UTF-16 crashes

Hi! Tested with 0.21.1. When tsgolint finds characters such as "✿" or "🎖️", it crashes hard:

```
panic: Bad UTF-16 character offset. Line: 30, character: 182. [recovered, repanicked]

goroutine 23254 [running]:
sync.(*WaitGroup).Go.func1.1()
        /opt/hostedtoolcache/go/1.26.2/x64/src/sync/waitgroup.go:251 +0x45
panic({0xa71840?, 0x32c83c7c1840?})
        /opt/hostedtoolcache/go/1.26.2/x64/src/runtime/panic.go:860 +0x13a
github.com/microsoft/typescript-go/internal/scanner.ComputePositionOfLineAndUTF16Character({0x32c84bfaa000?, 0x66157b?, 0x32c839b1d888?}, 0x32c84bfaa000?, 0xb6, {0x32c838488000, 0x16dd}, 0x0)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/scanner/scanner.go:2688 +0x28d
github.com/microsoft/typescript-go/internal/scanner.GetECMAPositionOfLineAndUTF16Character({0x118ae28, 0x32c839b1d888}, 0x1e, 0xb6)
        /home/runner/work/tsgolint/tsgolint/typescript-go/internal/scanner/scanner.go:2631 +0x78
main.printDiagnostic({{0x569, 0x588}, {0xc77cca, 0x1a}, {{0xc79ad0, 0x1c}, {0xcc516c, 0x6d}, {0x0, 0x0}}, ...}, ...)
        /home/runner/work/tsgolint/tsgolint/cmd/tsgolint/main.go:255 +0x1e5
main.runMain.func3()
        /home/runner/work/tsgolint/tsgolint/cmd/tsgolint/main.go:512 +0x295
sync.(*WaitGroup).Go.func1()
        /opt/hostedtoolcache/go/1.26.2/x64/src/sync/waitgroup.go:258 +0x4a
created by sync.(*WaitGroup).Go in goroutine 1
        /opt/hostedtoolcache/go/1.26.2/x64/src/sync/waitgroup.go:238 +0x73
error: "tsgolint" exited with code 2
```

--- a/cmd/tsgolint/main.go
+++ b/cmd/tsgolint/main.go
@@ -83,7 +83,6 @@ import (
 
 	"github.com/microsoft/typescript-go/shim/ast"
 	"github.com/microsoft/typescript-go/shim/bundled"
-	"github.com/microsoft/typescript-go/shim/core"
 	"github.com/microsoft/typescript-go/shim/scanner"
 	"github.com/microsoft/typescript-go/shim/tspath"
 	"github.com/microsoft/typescript-go/shim/vfs/cachedvfs"
@@ -246,13 +245,7 @@ func printDiagnostic(d rule.RuleDiagnostic, w *bufio.Writer, comparePathOptions 
 	codeboxEndLine := min(diagnosticEndline+1, len(lineMap)-1)
 
 	codeboxStart := scanner.GetECMAPositionOfLineAndUTF16Character(d.SourceFile, codeboxStartLine, 0)
-	var codeboxEndColumn int
-	if codeboxEndLine == len(lineMap)-1 {
-		codeboxEndColumn = len(text) - int(lineMap[len(lineMap)-1])
-	} else {
-		codeboxEndColumn = int(lineMap[codeboxEndLine+1]-lineMap[codeboxEndLine]) - 1
-	}
-	codeboxEnd := scanner.GetECMAPositionOfLineAndUTF16Character(d.SourceFile, codeboxEndLine, core.UTF16Offset(codeboxEndColumn))
+	codeboxEnd := scanner.GetECMAEndLinePosition(d.SourceFile, codeboxEndLine) + 1
 
 	w.Write([]byte{' ', 0x1b, '[', '7', 'm', 0x1b, '[', '1', 'm', 0x1b, '[', '3', '8', ';', '5', ';', '3', '7', 'm', ' '})
 	w.WriteString(d.RuleName)
@@ -337,12 +330,12 @@ func printDiagnostic(d rule.RuleDiagnostic, w *bufio.Writer, comparePathOptions 
 
 		if diagnosticHighlightActive {
 			underlineEnd = lineTextEnd
-		} else if int(lineMap[line]) <= diagnosticStart && (line == len(lineMap) || diagnosticStart < int(lineMap[line+1])) {
+		} else if int(lineMap[line]) <= diagnosticStart && (line == len(lineMap)-1 || diagnosticStart < int(lineMap[line+1])) {
 			underlineStart = min(max(lineTextStart, diagnosticStart), lineTextEnd)
 			underlineEnd = lineTextEnd
 			diagnosticHighlightActive = true
 		}
-		if int(lineMap[line]) <= diagnosticEnd && (line == len(lineMap) || diagnosticEnd < int(lineMap[line+1])) {
+		if int(lineMap[line]) <= diagnosticEnd && (line == len(lineMap)-1 || diagnosticEnd < int(lineMap[line+1])) {
 			underlineEnd = min(max(underlineStart, diagnosticEnd), lineTextEnd)
 			diagnosticHighlightActive = false
 		}

